api/wasm_test.go

```diff
@@ -139,3 +139,122 @@ func TestEncodeCastI64(t *testing.T) {
 		})
 	}
 }
+
+func TestEncodeDecodeI8x16(t *testing.T) {
+	for _, v := range [][2]uint64{
+		{0, 0},
+		{0, 0xffffffff_ffffffff},
+		{0xffffffff_ffffffff, 0},
+		{0xffffffff_ffffffff, 0xffffffff_ffffffff},
+		{0xffffffff_efffffff, 0xffffffff_ffffffff},
+		{0xffffffff_efffffff, 0xffffffff_efffffff},
+		{0xffff_ffff, 0xffff_ffff},
+		{0x1 << 4, 0x1 << 3},
+	} {
+		v := v
+		t.Run(fmt.Sprintf("%x", v), func(t *testing.T) {
+			low, hi := EncodeI8x16(DecodeI8x16(v[0], v[1]))
```

Comment:
Related to my other comment, it may be worth having tests that create slices and call `Encode` directly to confirm UX for them. We've confirmed the UX of passing result of Decode into Encode but the unit tests can be good confirmation / documentation of expected usage patterns as well.

Reply:
on it!